lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -1152,29 +1174,57 @@ LinearLayout TritonGPUDialect::toLinearLayout(ArrayRef<int64_t> shape,
   // To add a new layout add an else-if clause
   LinearLayout result = LinearLayout::empty();
   if (auto distributed = dyn_cast<DistributedEncodingTrait>(layout)) {
+    assert(allocationShape.empty() &&
+           "allocationShape not supported for distributed layout");
     result = distributed.toLinearLayout(shape);
   } else {
+    assert(!allocationShape.empty() &&
+           "allocationShape not supported for shared layout");
+    // We allow the first dimension to not be a power of 2 to model pipelining
+    if (!llvm::isPowerOf2_32(allocationShape.front()) ||
+        allocationShape.size() > shape.size()) {
+      allocationShape = allocationShape.drop_front();
+    }
+    // We just allow memdesc_views of at most one dimension larger than shape
+    assert(
+        allocationShape.size() == shape.size() &&
+        "allocationShape must be the same size or at most one larger as shape");
+    assert(llvm::all_of(allocationShape,
+                        [](int64_t dim) {
+                          return llvm::isPowerOf2_32(dim) && dim >= 1;
+                        }) &&
+           "allocationShape must be a postive power of 2");
+    assert(llvm::all_of(llvm::zip(allocationShape, shape),
+                        [](auto dims) {
+                          return std::get<0>(dims) >= std::get<1>(dims);
+                        }) &&
+           "allocationShape must be at least as large as shape");
+
```

Comment:
I wonder if it would be cleaner to split the op first so that you don't have to do all this weird logic and the toLinearLayout can just continue to take just one shape and one layout. 
This way the allocation shape would always be power of 2 and you wouldn't have to differentiate the shapes that comes from multibuffering from the others

Reply:
Discussed offline. The current approach looks good.